### PR TITLE
Feature | SER-2519 install and configure Matomo web analytics

### PR DESCRIPTION
**Pull request by suwiringSIxbert**
## Description
This Pull Request works towards integrating Matomo as one of the tools of the web frontend.
It handles tracking user and their actions through `pageviews` and `events`

## Changes Included in PR
- [SER-2078 - Integrate GrowthBook and Matomo Analytics with Fineract Mifos v2](https://oneacrefund.atlassian.net/browse/SER-2078)
- [SER-2519 - Install and configure Matomo web analytics](https://oneacrefund.atlassian.net/browse/SER-2519)

## PR Checklist
- [ ] Xray tests created
- [ ] PM tested and approved
- [ ] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`

### CODE REVIEW

**Review by amydoxym — APPROVED** [review]

**Comment by amydoxym** [comment]
@suwiringSIxbert Can we have you in one of our hangout sessions to explain to us how this Matomo works and what events/actions should we push to it (in case we are adding new features on the UI)
cc @gilclefes

**Comment by suwiringSIxbert** [comment]
> @suwiringSIxbert Can we have you in one of our hangout sessions to explain to us how this Matomo works and what events/actions should we push to it (in case we are adding new features on the UI) cc @gilclefes

@gilclefes Yes most definitely, let's put down something on the calendar for when y'all got a moment.

**Comment by suwiringSIxbert** [comment]
@amydoxym / @gilclefes any particular reason why the Azure build step is failing? I can't decipher the build logs

**Comment by gilclefes** [comment]
> @amydoxym / @gilclefes any particular reason why the Azure build step is failing? I can't decipher the build logs... I only added one lib `"ngx-matomo": "^1.1.0"`. Also, are environment variables passed to docker through CD or .yaml?

Not sure about the error as well currently, mostly adding libraries should not be an issue as I recently added some libraries which did not cause any build error. one thing though might be the angular cli version which is still 13 in docker even though in package json it is 14. I have updated that in my PR, i guess once that is merged we can rebuild this again.

**Comment by suwiringSIxbert** [comment]
@gilclefes I also just did an upgrade to 14 in Dockerfile but that doesn't seem to resolve the issue. Perhaps `ngx-matomo` is the culprit... Is it possible to upgrade to Angular 16? The latest version of `ngx-matomo` (i.e. 2.0.0) requires ng@16

File: src/environments/environment.ts [hunk 1]
```diff
@@ -32,6 +32,9 @@ export const environment = {
   defaultLanguage: window['env']['defaultLanguage'] || 'en-US',
   supportedLanguages: window['env']['supportedLanguages'] || 'en-US,fr-FR',
   headOfficeID: window['env']['headOfficeID'] || '1',
+  //Matomo instance config
+  matomoSiteId: window['env']['matomoSiteId'] || 1,
+  matomoSiteUrl: window['env']['matomoSiteUrl'] || 'https://analytics.test.oneacrefund.org',
```
Comment (Yann-J) [inline src/environments/environment.ts:37]: I would prefer if these were deployment-time env vars... Does this web app not support deploy-time configuration, or does it have to be compile-time?

**Review by amydoxym — APPROVED** [review]

**Review by Yann-J — APPROVED** [review]

File: src/environments/environment.ts (hunk 1 above)
Comment (gilclefes) [reply src/environments/environment.ts:37]: Not really sure if we support deploy-time for the web apps, @shejasabin @amydoxym  any ideas.

**Pull request merged**